kamelet-support/src/main/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteFileProcessor.java

```diff
@@ -0,0 +1,315 @@
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        StringParameter parsedParameter = new StringParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null),
+                null
+        );
+
+        return parsedParameter;
+    }
+
+    private NumberParameter getNumberParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        //IntegerParameter doesn't fit some long types there are present
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        NumberParameter parsedParameter = new NumberParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Long.parseLong(parameterData
+                        .getOrDefault("defaultValue", "0")
+                        .replaceAll("L", ""))
+        );
+
+        return parsedParameter;
+    }
+
+    private ObjectParameter getObjectParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        ObjectParameter parsedParameter = new ObjectParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                parameterData.getOrDefault("defaultValue", null)
+        );
+
+        return parsedParameter;
+    }
+
+    private BooleanParameter getBooleanParameter(
+            final Map.Entry<String, Map<String, String>> parameter) {
+        String id = parameter.getKey();
+        Map<String, String> parameterData = parameter.getValue();
+
+        BooleanParameter parsedParameter = new BooleanParameter(
+                id,
+                parameterData.get("displayName"),
+                parameterData.get("description"),
+                Boolean.parseBoolean(parameterData
+                        .getOrDefault("defaultValue", "false"))
+        );
+
+        return parsedParameter;
+    }
+
+    private String getStepType(final JsonObject component) {
+        final boolean canBeSource = component.getBoolean("producerOnly");
+        final boolean canBeSink = component.getBoolean("consumerOnly");
+
+        //If both are true or false we assume it to be an action.
+        //Or else, it has to be one of source or sink depending
+        //Take care that this later has to be duplicated because
+        //of the differences in how Camel components and kaoto components are
+        return canBeSource == canBeSink ? ACTION_TYPE
+                : canBeSource ? SOURCE_TYPE
+                : SINK_TYPE;
```

Comment:
Why? We have no domain over the integrated data and we could do nothing with that error. But having it could break the integration, over something that we really can't do "much". I could log it, but at this stage I can't see what info would be beneficial, instead of just adding a lot of noise maybe.

Reply:
I decided to log the error, and discard the component if this happens.